app/routes/api/maintainer.ts

```diff
@@ -0,0 +1,56 @@
+import { ActionFunction, json } from "@remix-run/node";
+import { getUserByUid } from "~/database.server";
+import { getProjectsMaintainedByUser } from "~/projects.server";
+
+/**
+ * This API endpoint verifies that a user is listed as the maintainer of at
+ * least 1 project in OSH. It expects a `userId` in the payload and a valid
+ * bearer token.
+ */
+export const action: ActionFunction = async ({ request }) => {
+  // Check that we have a valid userId in the payload
+  const data = await request.json();
+  const { userId } = data;
+  if (typeof userId !== "string") {
+    return new Response(null, {
+      status: 500,
+      statusText: "Malformed request",
+    });
+  }
+
+  // Check that we have a valid API key
+  const authHeader = request.headers.get("Authorization");
+  if (authHeader?.startsWith("Bearer ")) {
+    const receivedToken = authHeader.substring("Bearer ".length);
+
+    if (receivedToken !== process.env.BOT_API_KEY) {
```

Comment:
Before I even consider if I have security concerns here, is this endpoint basically just exposing public information via an API? If so, I'm not going to worry about it.

Reply:
It takes a Discord user and returns `true` if they have uploaded a project to OSH. The projects themselves are public information, but the association between Discord users and OSH projects feels private to me.